src/shims/unix/linux/eventfd.rs

```diff
@@ -150,7 +157,8 @@ impl FileDescription for Event {
         // types for current file description.
         ecx.check_and_update_readiness(self_ref)?;
 
-        Ok(Ok(U64_ARRAY_SIZE))
+        let result = Ok(U64_ARRAY_SIZE);
+        ecx.return_written_byte_count_or_error(result, dest)
```

Comment:
Why is `return_written_byte_count_or_error` a helper function? Every single caller fixes either `Err` or `Ok` so this is funneling through a `Result` without a good reason, it just makes the code harder to follow.

If there is any helper we want here, it is a new very generic helper that sets the last errors and also writes `-1` to `dest`. But that's probably better done in a future PR, for now you can just copy those 2 lines here I would say, like we do everywhere else.